Sources/WhoopDIKit/Module/AccumulationDefinition.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+private func wrapAccumulationProvider<Key: AccumulationKey>(name: String?, accumulationKey: Key.Type, valueProvider: @escaping (Any?) throws -> Key.AccumulatedValue) -> (Any?, Container) throws -> Any {
+    return { (params: Any?, container: Container) throws -> Any in
+        // Accumulate from the parent value if it exists, then add the current one
+        let previousValueFromContainer: Key.FinalValue?
+        do {
+            if let parent = container.parent {
```

Comment:
1. Let's try and extract this out of here so we don't need to expose the parent outside of container. 
2. I think it is generally dangerous capturing the parent within what is effectively the dependency definition (via the closure). This will hold a strong reference to the container (I think) and more importantly I don't think this traversal is always valid since it is set at the time the module is defined. Hard to explain in words, but this can then break during a local inject if this is a dependency of a dependency (see https://github.com/WhoopInc/WhoopDI/commit/aabea0ac90e6e5955d66124ffebc5d483c363f56).

Reply:
discussed offline. Added a test for this case and also passed parent in directly